### PR TITLE
Create and style basic recipe suggestions page

### PR DESCRIPTION
**Pull request by juliakimseohyeon**
Hi team, I was unfortunately unable to connect to the back-end and make API requests due to missing environmental variables... But I hardcoded in the data for now, and the basic style is ready for when the backend is connected. Thank you!

### CODE REVIEW

**Comment by nicmart-dev** [comment]
Hey Julia, thanks for getting a crack at this! Can you please resolve merge conflicts before review?

**Comment by juliakimseohyeon** [comment]
@nicmart-dev Git conflict has been addressed :) Thank you

File: server/controllers/recipesController.js [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+  const { prompt } = req.body; // Receive ingredients list from the front-end
 
-    try {
-        // URL of N8n webhook per https://docs.n8n.io/integrations/builtin/core-nodes/n8n-nodes-base.webhook/?utm_source=n8n_app&utm_medium=node_settings_modal-credential_link&utm_campaign=n8n-nodes-base.webhook
-        const webhookUrl = `${process.env.N8N_WEBHOOK_URL}/recommend-recipes`;
-        const response = await axios.post(webhookUrl, { ingredients });
+  try {
+    // URL of N8n webhook per https://docs.n8n.io/integrations/builtin/core-nodes/n8n-nodes-base.webhook/?utm_source=n8n_app&utm_medium=node_settings_modal-credential_link&utm_campaign=n8n-nodes-base.webhook
+    //const webhookUrl = `${process.env.N8N_WEBHOOK_URL}/recommend-recipes`;
+    const webhookUrl = process.env.N8N_WEBHOOK_URL_RECIP;
```
Comment (nicmart-dev) [inline server/controllers/recipesController.js]: See my comment to Jacob https://github.com/nicmart-dev/feedmenow/pull/36#discussion_r1697715660

**Review by nicmart-dev — CHANGES_REQUESTED** [review]
Hey Julia, there is pending conflict still, can you please resolve this and I'll have a look. 
Also regarding your original post comment, can you try to get api keys using the [new wiki page](https://github.com/nicmart-dev/feedmenow/wiki/Secrets-environment-variables-using-GitHub-Actions-workflow-webhook#how-to-get-github-secrets) and see if you can replace hardcoded data?

**Comment by juliakimseohyeon** [comment]
@nicmart-dev It seems the new conflicts emerged after I resolved the first set of conflicts, due to newest changes being pushed to develop. Please take another look when you have a moment, thanks!

**Comment by nicmart-dev** [comment]
Indeed conflict resolved. My other points from my last comment are still pending though :)

**Comment by juliakimseohyeon** [comment]
Hi @nicmart-dev, hoping to get a little guidance on this as I’ve never used the webhook before and am a bit confused. I got to the part where I open my getAPI request on webhook, but the request appears empty. How do I go about updating the workflow with a new URL from here?
Screenshot 2024-07-30 at 6 38 00 PM

**Comment by nicmart-dev** [comment]
It's the POST request just below that GET

**Comment by yaacovkdev** [comment]
Merged recipe suggestions page. The Server API key is set to local, due to cloud issues on the global n8n, so the root url on .env in server has to be changed.

**Review by yaacovkdev — COMMENTED** [review]
The branches are merged

**Comment by nicmart-dev** [comment]
> Merged recipe suggestions page. The Server API key is set to local, due to cloud issues on the global n8n, so the root url on .env in server has to be changed.

what is still the issue? I thought you eventually managed to connect since we spoke?

**Comment by nicmart-dev** [comment]
This is still an issue https://github.com/nicmart-dev/feedmenow/pull/36#discussion_r1697715660

File: server/server.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const envConfig = dotenv.config();
 const dotenvExpand = require("dotenv-expand");
 dotenvExpand.expand(envConfig);
 
-const PORT = process.env.PORT || 5000; // Define the port number, use environment variable if available
```
Comment (nicmart-dev) [inline server/server.js]: I suggest to keep fallback as 5000 as you can always change it on your local .env

File: .idea/modules.xml
Comment (nicmart-dev) [inline .idea/modules.xml:1]: What are all those files in this new `.idea` folder? If not needed, please remove them.

File: client/src/pages/RecipeSuggestions.jsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            imageAlt: 'recipe image',
+            imageSrc:
+                'https://images.unsplash.com/photo-1670398564097-0762e1b30b3a?q=80&w=2071&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
+            href: 'www.fakeurl.com',
+            name: <FormattedMessage id="recipes.name" />,
+            time: '30 min',
+            ingredient: '5 ingredients',
+            cuisine: 'French',
+        },
+    ]
```
Comment (nicmart-dev) [inline client/src/pages/RecipeSuggestions.jsx]: This `recipes` variables seems like placeholder and should be removed as likely replaced by output from n8n.

File: server/controllers/recipesController.js [hunk 4]
```diff
@@ -5,32 +5,25 @@ const jwt = require("jsonwebtoken");
 // POST /api/recipes/suggest
 const suggestRecipes = async (req, res) => {
     const { ingredients } = req.body; // Receive ingredients list from the front-end
-
+    console.log("sup???");
```
Comment (nicmart-dev) [inline server/controllers/recipesController.js]: Likely used for testing only and should be removed

**Review by nicmart-dev — CHANGES_REQUESTED** [review]
Good work on the the recipes suggestions page Jacob! I have yet to test the flow but have provided some preliminary comments after reviewing the code changes.

I see the branch is [13 commits behind](https://github.com/nicmart-dev/feedmenow/compare/feature/frontend-recipe-suggestions...develop) so may need to merge from develop branch.

Very weird that my non reviewed [changes](https://github.com/nicmart-dev/feedmenow/pull/35/commits) to the user preferences page (which are part of another branch and draft PR) made it to this PR though. I don't think we need to undo them now but it does complicate things a bit with this review :)

File: server/controllers/recipesController.js [hunk 5]
```diff
@@ -4,33 +4,54 @@ const jwt = require("jsonwebtoken");
...
-  };
+    const { ingredients } = req.body; // Receive ingredients list from the front-end
 
   try {
     // URL of N8n webhook per https://docs.n8n.io/integrations/builtin/core-nodes/n8n-nodes-base.webhook/?utm_source=n8n_app&utm_medium=node_settings_modal-credential_link&utm_campaign=n8n-nodes-base.webhook
     //const webhookUrl = `${process.env.N8N_WEBHOOK_URL}/recommend-recipes`;
-    const webhookUrl = process.env.N8N_WEBHOOK_URL_RECIP_TEST;
+    const webhookUrl = process.env.N8N_WEBHOOK_URL_RECIPE;
```
Comment (nicmart-dev) [inline server/controllers/recipesController.js]: Not sure why the new env variable which also is not found in https://github.com/nicmart-dev/feedmenow/blob/develop/server/example.env
I recommend to revert this change and restore line 11

Comment (yaacovkdev) [reply server/controllers/recipesController.js]: Sorry, I created a few more server env variables to send requests to the testing route on n8n. Line 11 can be revoked to its previous state.

**Comment by yaacovkdev** [comment]
.idea files have been removed. Thank you Nic.

File: server/controllers/recipesController.js (hunk 5 above)
Comment (nicmart-dev) [reply server/controllers/recipesController.js]: Was that fixed?

**Comment by nicmart-dev** [comment]
Is it now fully working with n8n.nicmart.synology.me?

Also can you please address some lf the outstanding comments and resolve corresponding threads if fixed?

File: client/src/pages/RecipeSuggestions.jsx (hunk 3 above)
Comment (nicmart-dev) [reply client/src/pages/RecipeSuggestions.jsx]: This appears to be still outstanding.

Comment (nicmart-dev) [reply client/src/pages/RecipeSuggestions.jsx]: Still outstanding

File: server/server.js (hunk 2 above)
Comment (nicmart-dev) [reply server/server.js]: Still outstanding

File: client/src/pages/RecipeSuggestions.jsx (hunk 3 above)
Comment (yaacovkdev) [reply client/src/pages/RecipeSuggestions.jsx]: The placeholder recipes have been removed

File: server/server.js (hunk 2 above)
Comment (yaacovkdev) [reply server/server.js]: The port has been changed back to 5000.

File: server/controllers/recipesController.js (hunk 1 above)
Comment (yaacovkdev) [reply server/controllers/recipesController.js]: The .env has been updated and proper values placed.

**Comment by nicmart-dev** [comment]
@yaacovkdev great work on getting this to work end to end from entering recipe in UI to running n8n workflow to display recipes as output.
Things I noticed to address separately as bug fixes or individual enhancements:
#39
#40 

Also it would be great to update the [doc](https://github.com/nicmart-dev/feedmenow/wiki/n8n-workflow) to reflect the latest n8n workflow
{02EE0067-E91D-4D7F-996D-5C2809FC7C0B}
